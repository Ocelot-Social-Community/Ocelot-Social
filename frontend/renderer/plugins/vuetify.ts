// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
  })
