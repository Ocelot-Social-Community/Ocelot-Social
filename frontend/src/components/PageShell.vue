<template>
  <v-app>
    <TopMenu />
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
