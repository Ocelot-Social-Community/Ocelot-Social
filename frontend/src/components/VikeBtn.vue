<template>
  <v-btn
    :variant="isRouteSelected($attrs.href as string) ? 'tonal' : 'flat'"
    @click.prevent="onClick($attrs.href as string)"
  >
    <slot />
  </v-btn>
</template>
<script lang="ts" setup>
import { navigate } from 'vike/client/router'

import { usePageContext } from '#context/usePageContext'

const pageContext = usePageContext()

function onClick(href: string) {
  return navigate(href)
}

const isRouteSelected = (href: string) => {
  if (href === '/app') {
    return pageContext.urlPathname.indexOf(href) === 0
  }
  return pageContext.urlPathname === href
}
</script>
