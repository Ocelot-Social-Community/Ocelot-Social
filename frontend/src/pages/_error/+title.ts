export const title = 'IT4C | Error'
