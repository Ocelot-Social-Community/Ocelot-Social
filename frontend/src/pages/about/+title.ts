export const title = 'IT4C | About'
