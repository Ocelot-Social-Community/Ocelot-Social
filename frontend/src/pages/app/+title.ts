export const title = 'IT4C | App'
