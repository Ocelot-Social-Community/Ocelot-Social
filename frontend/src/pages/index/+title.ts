import { META } from '#src/env'

export const title = META.DEFAULT_TITLE
