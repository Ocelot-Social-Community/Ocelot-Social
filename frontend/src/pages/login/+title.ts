export const title = 'Ocelot - Login'
