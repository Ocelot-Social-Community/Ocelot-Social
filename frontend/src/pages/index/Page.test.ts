import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import IndexPage from './+Page.vue'
import { title } from './+title'

describe('IndexPage', () => {
  const wrapper = mount(VApp, {
    slots: {
      default: h(IndexPage as Component),
    },
  })

  it('title returns default title', () => {
    expect(title).toBe('IT4C')
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
