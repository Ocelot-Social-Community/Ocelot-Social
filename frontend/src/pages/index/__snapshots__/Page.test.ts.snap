// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IndexPage > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    <main
      class="v-main bg-grey-lighten-3"
      style="--v-layout-left: 0px; --v-layout-right: 0px; --v-layout-top: 0px; --v-layout-bottom: 0px;"
    >
      
      <div
        class="v-container v-locale--is-ltr"
      >
        <div
          class="v-row"
        >
          <!--v-if-->
          <div
            class="v-col"
          >
            <div
              class="v-sheet v-theme--light v-sheet--rounded pa-3"
            >
              
              <h1>
                $t('home.h1')
              </h1>
              <p>
                $t('home.text1')
              </p>
              <br />
              <p>
                $t('home.text2')
              </p>
              <br />
              <p>
                $t('home.text3')
              </p>
              <br />
              <p>
                $t('home.text4') 
                <i
                  aria-hidden="true"
                  class="mdi-heart mdi v-icon notranslate v-theme--light v-icon--size-default text-red"
                />
              </p>
              <br />
              <p>
                $t('home.greet1')
              </p>
              <p>
                $t('home.greet2')
              </p>
              
            </div>
          </div>
        </div>
      </div>
      
    </main>
    
  </div>
</div>
`;
